File: apps/player-android/app/src/main/java/org/tilecast/player/content/CacheIdentity.kt
package org.tilecast.player.content

import org.tilecast.player.core.ServerUrlPolicy

data class CacheIdentity(val installationId: String, val screenId: String, val normalizedServerUrl: String)

internal fun cacheIdentity(serverUrl: String, installationId: String?, screenId: String?): CacheIdentity? {
    if (installationId.isNullOrBlank() || screenId.isNullOrBlank()) return null
    val normalized = ServerUrlPolicy.normalize(serverUrl).getOrNull()?.value ?: return null
    return CacheIdentity(installationId, screenId, normalized)
}

internal fun CacheIdentity.matches(installationId: String?, screenId: String?, normalizedServerUrl: String?): Boolean =
    installationId == this.installationId && screenId == this.screenId && normalizedServerUrl == this.normalizedServerUrl

File: apps/player-linux/src/core/cache-identity.ts
import { normalizeServerUrl } from "./server-url";

export interface CacheIdentity {
  installationId: string;
  screenId: string;
  normalizedServerUrl: string;
}

export function makeCacheIdentity(
  serverUrl: string,
  installationId: string | null | undefined,
  screenId: string | null | undefined,
): CacheIdentity | null {
  const normalized = normalizeServerUrl(serverUrl);
  if (!normalized.ok || !normalized.url || !installationId || !screenId) {
    return null;
  }
  return { installationId, screenId, normalizedServerUrl: normalized.url };
}

export function cacheIdentityMatches(
  identity: CacheIdentity,
  value: Partial<CacheIdentity> | null | undefined,
): boolean {
  return (
    value?.installationId === identity.installationId &&
    value?.screenId === identity.screenId &&
    value?.normalizedServerUrl === identity.normalizedServerUrl
  );
}
